pkg/deploy/generator/resources_rp.go

```diff
@@ -916,16 +915,16 @@ done
 	}
 }
 
+func (g *generator) rpParentDNSZone() *arm.Resource {
+	return g.dnsZone("[parameters('rpParentDomainName')]")
+}
+
+func (g *generator) rpClusterParentDNSZone() *arm.Resource {
+	return g.dnsZone("[parameters('clusterParentDomainName')]")
+}
+
```

Comment:
:exploding_head: abstractions x 3 times :D

Reply:
ech, it's not perfect but I think it's good enough.  No-one will feel bad about changing it.